src/image-mime.ts

```diff
@@ -18,13 +18,15 @@ export function isImageMimeType(str: string): str is ImageMime {
  * Throws an Error is if extension doesn't correspond to an {@link ImageMime}.
  */
 export function getImageMimeType(href: string): ImageMime {
-  if (href.endsWith(".png")) {
+  const src = href.split("?")[0];
```

Comment:
Instead of manually splitting, I'd prefer a true parse. Could you swap to something like:
```ts
const src = new URL(href).pathname
```

Reply:
Hi Erik, I understand your concern. Even though URL.pathname looks more elegant, but it only works for absolute URL paths, for those ones like "test.jpg", "blah/foo.jpeg"... which I took from `image-mime.test.ts` and some relative path like "/wp-content/uploads/2020/10/Google-Workspace-Icons.png?w=680", it always throws Invalid URL error. Please consider this, thank you for responding!

Reply:
To avoid the Invalid URL error on relative URLs, we can set a base string in `new URL()`, like `new URL(href, 'https://aaa.com').pathname`, but this makes the source code weirder than slipt() I think.

Reply:
Yeah, I think it's the right call. For style, I would still slightly change it to:
```
const [src] = href.split("?", 1);
```

Reply:
Sure, you're the boss :D Would you please review the latest change, thank you!

<img width="651" alt="Screenshot 2024-04-12 at 10 24 43 PM" src="https://github.com/erikbrinkman/teapub/assets/795334/100d3083-b4a2-4af6-b493-11d30c4243a3">